homeassistant/components/homewizard_energy/sensor.py

```diff
@@ -0,0 +1,228 @@
+"""Creates Homewizard Energy sensor entities."""
+from __future__ import annotations
+
+import logging
+from typing import Final
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.const import (
+    CONF_ID,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_SIGNAL_STRENGTH,
+    DEVICE_CLASS_TIMESTAMP,
+    ENERGY_KILO_WATT_HOUR,
+    PERCENTAGE,
+    POWER_WATT,
+    VOLUME_CUBIC_METERS,
+)
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ATTR_ACTIVE_POWER_L1_W,
+    ATTR_ACTIVE_POWER_L2_W,
+    ATTR_ACTIVE_POWER_L3_W,
+    ATTR_ACTIVE_POWER_W,
+    ATTR_GAS_TIMESTAMP,
+    ATTR_METER_MODEL,
+    ATTR_SMR_VERSION,
+    ATTR_TOTAL_ENERGY_EXPORT_T1_KWH,
+    ATTR_TOTAL_ENERGY_EXPORT_T2_KWH,
+    ATTR_TOTAL_ENERGY_IMPORT_T1_KWH,
+    ATTR_TOTAL_ENERGY_IMPORT_T2_KWH,
+    ATTR_TOTAL_GAS_M3,
+    ATTR_WIFI_SSID,
+    ATTR_WIFI_STRENGTH,
+    CONF_API,
+    CONF_DATA,
+    CONF_MODEL,
+    CONF_SW_VERSION,
+    COORDINATOR,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS: Final[tuple[SensorEntityDescription, ...]] = (
+    SensorEntityDescription(
+        key=ATTR_SMR_VERSION,
+        name="SMR Version",
+        icon="mdi:wifi",
+    ),
+    SensorEntityDescription(
+        key=ATTR_METER_MODEL,
+        name="Model",
+        icon="mdi:counter",
+    ),
+    SensorEntityDescription(
+        key=ATTR_WIFI_SSID,
+        name="Wifi SSID",
+        icon="mdi:wifi",
+    ),
+    SensorEntityDescription(
+        key=ATTR_WIFI_STRENGTH,
+        name="Wifi Strength",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=DEVICE_CLASS_SIGNAL_STRENGTH,
```

Comment:
percent is an invalid unit for device_class signal_strength
https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
What do you think that is the best we can do? As the valid by the API is in percentage. Just remove `device_class`?

Reply:
I'd remove device_class and configure a nice icon. 
Maybe it can even be converted 🤔

Reply:
Agreed, I've removed device_class in 2ad596de7f9b5ff0b647f306b4af11631a388d71. Maybe in the future make a conversion to the correct unit.